addons/delivery/models/delivery_carrier.py

```diff
@@ -108,6 +108,15 @@ def _match_address(self, partner):
             return False
         return True
 
+    def _get_delivery_type(self):
+        ''' Hook to send the delivery type
+        
+        This function needs to be overridden by a delivery carrier if it implements 
+        delivery subtypes
+        '''
+        self.ensure_one()
+        return self.delivery_type
+
```

Comment:
You shoudln't put a hook for another module before all the onchanges functions IMO, there should be a better place in the file for this